### PR TITLE
Feat/devnet no mining

### LINKED ISSUE #414: Add `--no-mining` option to `deps` Devnet and Cartesi Rollups Cli
## 📚 Context

In order to have better control on manual and end to end tests, a better control over block generation is needed

## ✔️ Solution

Add the Devnet supported `--no-mining` flag to `deps`package and Cartesi Rollups Cli

### PR DESCRIPTION
**Pull request by fmoura**
Closes #414

### CODE REVIEW

File: internal/deps/deps.go [hunk 1]
```diff
@@ -206,12 +209,22 @@ func Run(ctx context.Context, depsConfig DepsConfig) (*DepsContainers, error) {
...
+		var waitStrategy *wait.LogStrategy
+		if depsConfig.Devnet.NoMining {
+			cmd = []string{"anvil", "--no-mining",
+				"--load-state", "/usr/share/devnet/anvil_state.json"}
+			waitStrategy = wait.ForLog("net_listening")
+		} else {
+			cmd = []string{"anvil", "--block-time",
+				depsConfig.Devnet.BlockTime, "--load-state", "/usr/share/devnet/anvil_state.json"}
+			waitStrategy = wait.ForLog("Block Number: " + depsConfig.Devnet.BlockToWaitForOnStartup)
+		}
```
Comment (marcelstanley) [inline internal/deps/deps.go:222]:
```suggestion
		cmd := []string{
             "anvil",
             "--load-state",
             "/usr/share/devnet/anvil_state.json",
         }
         var waitStrategy *wait.LogStrategy
         if depsConfig.Devnet.NoMining {
             cmd = append(cmd, "--no-mining")
             waitStrategy = wait.ForLog("net_listening")
         } else {
             cmd = append(cmd, "--block-time",
                 depsConfig.Devnet.BlockTime)
             waitStrategy = wait.ForLog("Block Number: " + depsConfig.Devnet.BlockToWaitForOnStartup)
         }
```

File: cmd/cartesi-rollups-cli/root/mine/mine.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	Cmd.Flags().StringVar(&anvilEndpoint, "anvil-endpoint", "http://localhost:8545",
+		"anvil address used to send to the request")
+}
+
+func run(cmd *cobra.Command, args []string) {
+
+	cobra.CheckErr(ethutil.MineNewBlock(context.Background(), anvilEndpoint))
+
+	slog.Info("Ok")
```
Comment (marcelstanley) [inline cmd/cartesi-rollups-cli/root/mine/mine.go:38]: Perhaps we could enrich the logs saying the block number that was mined.

**Review by marcelstanley — COMMENTED** [review]
Nice work!!

I have a few suggestions, though.

Also, I'm wondering whether:

1. We should add a new test to `pkg/ethutil/ethutil_test.go` to account for new function `MineNewBlock()`.

2. Refactor the two pieces of code below to use `NewDefaultDepsConfig()` (in a separate PR, if needed):

https://github.com/cartesi/rollups-node/blob/cb7d11aee6f690e2aab0753e75e44dc8d8f79f51/internal/node/machinehash_test.go#L139-L144

https://github.com/cartesi/rollups-node/blob/cb7d11aee6f690e2aab0753e75e44dc8d8f79f51/pkg/ethutil/ethutil_test.go#L106-L111
- [ ]  in another task, refactor internal/node/machinehash_test.go

to use https://github.com/cartesi/rollups-node/blob/cb7d11aee6f690e2aab0753e75e44dc8d8f79f51/internal/deps/deps.go#L60:

File: internal/deps/deps.go [hunk 3]
```diff
@@ -27,6 +27,7 @@ const (
 	DefaultDevnetPort              = "8545"
 	DefaultBlockTime               = "1"
 	DefaultBlockToWaitForOnStartup = "21"
+	DefaultNoMining                = false
```
[the remark this answers is not in the document]
Comment (fmoura) [reply internal/deps/deps.go:30]: The ones aboive should get Devnet on their names as well

**Comment by fmoura** [comment]
> Nice work!!
> 
> I have a few suggestions, though.
> 
> Also, I'm wondering whether:
> 
> 1. We should add a new test to `pkg/ethutil/ethutil_test.go` to account for new function `MineNewBlock()`.
> 2. Refactor the two pieces of code below to use `NewDefaultDepsConfig()` (in a separate PR, if needed):
> 
> https://github.com/cartesi/rollups-node/blob/cb7d11aee6f690e2aab0753e75e44dc8d8f79f51/internal/node/machinehash_test.go#L139-L144
> 
> https://github.com/cartesi/rollups-node/blob/cb7d11aee6f690e2aab0753e75e44dc8d8f79f51/pkg/ethutil/ethutil_test.go#L106-L111
> 
> * [ ]   in another task, refactor internal/node/machinehash_test.go
> 
> to use
> 
> https://github.com/cartesi/rollups-node/blob/cb7d11aee6f690e2aab0753e75e44dc8d8f79f51/internal/deps/deps.go#L60
> 
> :

Here these two examples uses a Subset of the NewDepsConfig. A better solution would be have multiple "NewDefaultConfig" , one if you want to use just Devnet and other just for Postgres at least. It will fit better on another PR

**Review by marcelstanley — APPROVED** [review]

**Pull request merged**